R/robttbayesianmodelaveragedinternal.R

```diff
@@ -0,0 +1,1241 @@
+
+  modelsSummary$addColumnInfo(name = "number",              title = "#",                         type = "integer")
+  modelsSummary$addColumnInfo(name = "distribution",        title = gettext("Distribution"),     type = "string")
+  modelsSummary$addColumnInfo(name = "priorEffect",         title = gettext("Effect"),           type = "string", overtitle = overtitlePrior)
+  modelsSummary$addColumnInfo(name = "priorHeterogeneity",  title = gettext("Heterogeneity"),    type = "string", overtitle = overtitlePrior)
+  modelsSummary$addColumnInfo(name = "priorOutliers",       title = gettext("Outliers"),         type = "string", overtitle = overtitlePrior)
+  modelsSummary$addColumnInfo(name = "priorProb",           title = gettext("P(M)"),             type = "number")
+
+  for (i in 1:nrow(fitSummary[["summary"]])) {
+    tempRow <- list(
+      number             = fitSummary[["summary"]][i, "Model"],
+      distribution       = fitSummary[["summary"]][i, "Distribution"],
+      priorEffect        = fitSummary[["summary"]][i, "delta"],
+      priorHeterogeneity = fitSummary[["summary"]][i, "rho"],
+      priorOutliers      = fitSummary[["summary"]][i, "nu"],
+      priorProb          = fitSummary[["summary"]][i, "prior_prob"]
+    )
+
+    modelsSummary$addRows(tempRow)
+  }
+  modelsSummary$addFootnote(gettext("The analysis will estimate multiple models using MCMC and might require a prolonged time to complete."), symbol = "\u26A0")
+
+  modelPreview[["modelsSummary"]] <- modelsSummary
+
+
+  return()
+}
+.robttFitModel                 <- function(jaspResults, dataset, options) {
+
+  if (!is.null(jaspResults[["model"]]))
+    return()
+
+  model <- createJaspState()
+  model$dependOn(.robttDependencies)
+  jaspResults[["model"]] <- model
+
+
+  priors <- jaspResults[["priors"]]$object
+  fit    <- try(RoBTT::RoBTT(
+    # data
+    x1 = dataset[[options[["dependent"]]]][dataset[[options[["group"]]]] == levels(dataset[[options[["group"]]]])[1]],
+    x2 = dataset[[options[["dependent"]]]][dataset[[options[["group"]]]] == levels(dataset[[options[["group"]]]])[2]],
+
+    # priors
+    prior_delta      = priors[["modelsEffect"]],
+    prior_rho        = priors[["modelsHeterogeneity"]],
+    prior_nu         = priors[["modelsOutliers"]],
+    prior_delta_null = priors[["modelsEffectNull"]],
+    prior_rho_null   = priors[["modelsHeterogeneityNull"]],
+    prior_nu_null    = priors[["modelsOutliersNull"]],
+    # sampling settings
+    chains  = options[["advancedMcmcChains"]],
+    warmup  = options[["advancedMcmcAdaptation"]],
+    iter    = options[["advancedMcmcAdaptation"]] + options[["advancedMcmcSamples"]],
+    thin    = options[["advancedMcmcThin"]],
+    # additional settings
+    convergence_checks = RoBTT::set_convergence_checks(
+      max_Rhat            = 1.05,
+      min_ESS             = 500
+    ),
+    save     = "all",
+    seed     = options[["seed"]], # TODO: .getSeedJASP(options)
```

Comment:
todo?

Reply:
thanks, fixed!
(the option didn't work because of the R-version logic operations change bug)